pkg/machine/e2e/config_test.go

```diff
@@ -274,6 +275,18 @@ func isWSL() bool {
 	return isVmtype(define.WSLVirt)
 }
 
+// Only used on Windows
+//
+//nolint:unparam,unused
+func isSSHPortOpen() bool {
```

Comment:
I would not use `Open` here, as an open network port means it is possible to connect to it (something is listening, and no firewall blocks the connection). Maybe `Free` or `Available`?

Reply:
yes this name and check seems wrong. We should no try to connect as this might fail for other reasons.
If we want to know if the port is used or not we should just try to bind it.
```
listener, err := net.Listen("tcp", "0.0.0.0:2222")
Expect(err).ToNot(HaveOccurred())
defer listener.Close()
```